tooling/src/hypothesistooling/__main__.py

```diff
@@ -67,127 +71,64 @@ def lint():
-
-    # We time out after an hour, which is a stupidly long time and it should
-    # never actually take that long: A full Travis run only takes about 20-30
-    # minutes! This is really just here as a guard in case something goes
-    # wrong and we're not paying attention so as to not be too mean to Travis..
-    while time() <= start_time + 60 * 60:
-        jobs = tools.build_jobs()
-
-        failed_jobs = [
-            (k, v)
-            for k, vs in jobs.items()
-            if k not in PENDING_STATUS + ('passed',)
-            for v in vs
-        ]
-
-        if failed_jobs:
-            print('Failing this due to failure of jobs %s' % (
-                ', '.join('%s(%s)' % (s, j) for j, s in failed_jobs),
-            ))
-            sys.exit(1)
-        else:
-            pending = [j for s in PENDING_STATUS for j in jobs.get(s, ())]
-            try:
-                # This allows us to test the deploy job for a build locally.
-                pending.remove('deploy')
-            except ValueError:
-                pass
-            if pending:
-                still_pending = set(pending)
-                if prev_pending is None:
-                    print('Waiting for the following jobs to complete:')
-                    for p in sorted(still_pending):
-                        print(' * %s' % (p,))
-                    print()
-                else:
-                    completed = prev_pending - still_pending
-                    if completed:
-                        print('%s completed since last check.' % (
-                            ', '.join(sorted(completed)),))
-                prev_pending = still_pending
-                naptime = 10.0 * (2 + random.random())
-                print('Waiting %.2fs for %d more job%s to complete' % (
-                    naptime, len(pending), 's' if len(pending) > 1 else '',))
-                sleep(naptime)
-            else:
-                break
-    else:
-        print("We've been waiting for an hour. That seems bad. Failing now.")
-        sys.exit(1)
 
-    print('Looks good to release!')
+@task()
+def deploy():
+    print('Current head:', HEAD)
+    print('Current master:', MASTER)
 
-    if os.environ.get('TRAVIS_SECURE_ENV_VARS', None) != 'true':
-        print("But we don't have the keys to do it")
+    if not tools.is_ancestor(HEAD, MASTER):
+        print('Not deploying due to not being on master')
+        sys.exit(0)
+
+    if not tools.has_travis_secrets():
+        print('Running without access to secure variables, so no deployment')
         sys.exit(0)
 
     print('Decrypting secrets')
     tools.decrypt_secrets()
+    tools.configure_git()
 
-    print('Release seems good. Pushing to github now.')
-
-    tools.create_tag_and_push()
-
-    print('Now uploading to pypi.')
-
-    subprocess.check_call([
-        sys.executable, '-m', 'twine', 'upload',
-        '--config-file', tools.PYPIRC,
-        os.path.join(DIST, '*'),
-    ])
+    do_release(hp)
```

Comment:
Is it correct that a generic deploy() task does a hypothesis-python release? Most of the other stuff in this file seems to state a dependency on hypothesis-python changes.

Reply:
The `do_release` function checks if there are no changes and does nothing in that case. Once there is more than one project it will look like this:

```python
for project in tools.all_projects():
    do_release(project)
```

Maybe I should change it to that in this release too?